homeassistant/components/sensor/speedtest.py

```diff
@@ -0,0 +1,105 @@
+"""
+homeassistant.components.sensor.speedtest
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Speedtest.net sensor based on speedtest-cli.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.speedtest/
+"""
+import logging
+import sys
+import re
+from datetime import timedelta
+from subprocess import check_output
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['speedtest-cli==0.3.4']
+_LOGGER = logging.getLogger(__name__)
+
+_SPEEDTEST_REGEX = re.compile(r'Ping:\s(\d+\.\d+)\sms\nDownload:\s(\d+\.\d+)'
+                              r'\sMbit/s\nUpload:\s(\d+\.\d+)\sMbit/s\n')
+
+SENSOR_TYPES = {
+    'ping': ['Ping', 'ms'],
+    'download': ['Download', 'Mbit/s'],
+    'upload': ['Upload', 'Mbit/s'],
+}
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(hours=1)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Setup the Speedtest sensor. """
+
+    data = SpeedtestData(hass.config.path)
+
+    dev = []
+    for variable in config['monitored_conditions']:
+        if variable not in SENSOR_TYPES:
+            _LOGGER.error('Sensor type: "%s" does not exist', variable)
+        else:
+            dev.append(SpeedtestSensor(data, variable))
+
+    add_devices(dev)
+
+
+# pylint: disable=too-few-public-methods
+class SpeedtestSensor(Entity):
+    """ Implements a speedtest.net sensor. """
+
+    def __init__(self, speedtest_data, sensor_type):
+        self.client_name = 'Speedtest'
```

Comment:
This value appears static and only used in one location. That indicates that it may not need to be a property. This value could either be hard coded into the name property or defined as a constant at the top of the module. This seems like an odd place to scope it, though.